### PR TITLE
DAOS-2450 Test: Fix tag=quick such that it is a quick verification

### PR DESCRIPTION
**Pull request by omar-ocampo**
Added daos cont autotest subcommand to daos_utils.
    Command "daos cont autotest" is now tested within container/autotest.py.

Skip-unit-test: true
Test-tag-hw-small: pr,hw,small quick

Signed-off-by: Omar Ocampo <omar.ocampo.coronado@intel.com>

### CODE REVIEW

File: src/tests/ftest/container/autotest.py
Comment (berserk-fury) [inline src/tests/ftest/container/autotest.py:3]: not needed

File: src/tests/ftest/container/autotest.yaml
Comment (cibervicho) [inline src/tests/ftest/container/autotest.yaml:5]: Shouldn't this be "client-B" instead? Like this:

hosts:
    test_servers:
        - server-A
    test_clients:
        - client-B

**Review by berserk-fury — DISMISSED** [review]
The new `autotest` that this PR adds passed

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5251/5/artifact/Functional_Hardware_Small/container/autotest.py/test-results/1-._container_autotest.py_ContainerAutotestTest.test_container_autotest_hosts-pool-servers-f24f/debug.log

**Review by cibervicho — DISMISSED** [review]

File: src/tests/ftest/container/autotest.yaml
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.yaml:5]: Good catch, will fix

**Review by berserk-fury — APPROVED** [review]

**Review by cibervicho — APPROVED** [review]
Test passed:

[2021-04-09T04:47:54.450Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=quick --mux-yaml /tmp/tmp5_s5kzf4/container-autotest.yaml -- ./container/autotest.py
[2021-04-09T04:47:54.450Z] JOB ID     : b011133bc3eb36187f9ec3ab6409d625ca5ace46
[2021-04-09T04:47:54.450Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2021-04-09T04.47-b011133/job.log
[2021-04-09T04:52:31.527Z]  (1/1) ./container/autotest.py:ContainerAutotestTest.test_container_autotest;hosts-pool-servers-40f9:  PASS (272.61 s)
[2021-04-09T04:52:31.527Z] RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by omar-ocampo** [comment]
Error not related to this PR:
Test Hardware / Functional_Hardware_Medium / 01-./daos_test/daos_core_test-rebuild.py:DaosCoreTestRebuild.test_rebuild_0to10;transport_config-args-daos_test-stopped_ranks-test_name-transport_config-hosts-pool-0-1-transport_config-timeouts-c799 – FTEST_daos_test.DaosCoreTestRebuild

**Comment by omar-ocampo** [comment]
Test passing: https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-5251/runs/7/nodes/1342/log/?start=0

File: src/tests/ftest/container/autotest.py
Comment (shimizukko) [inline src/tests/ftest/container/autotest.py:22]: You don't really need these variables. You can just make them local.

File: src/tests/ftest/container/autotest.py
Comment (shimizukko) [inline src/tests/ftest/container/autotest.py:34]: I'm not familiar with this command, but what happens if the test fails?

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:22]: Ok

File: src/tests/ftest/container/autotest.py
Comment (shimizukko) [reply src/tests/ftest/container/autotest.py:34]:
```suggestion
try:
    self.daos_cmd.pool_autotest(pool=self.pool.uuid)
    self.log.info("daos pool autotest passed.")
except CommandFailure as error:
    self.fail("daos pool autotest failed!")
```
I see. Then you can do something like this to make the output cleaner.

File: src/tests/ftest/container/autotest.py
Comment (shimizukko) [inline src/tests/ftest/container/autotest.py:28]: I mean you don't need these variables at all. You also don't need the `__init__()`.

**Review by shimizukko — APPROVED** [review]

**Comment by sylviachanoiyee** [comment]
@omar-ocampo , could you look into comments from Makito, and rerun the PR without skipping any tests?
Remove all CI commit pragma and just use the one below
Test-tag-hw-small: pr,hw,small autotest

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py]: Code removed, in local run it shown no changes.

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py]: The exception was imported and used as suggested.

**Comment by omar-ocampo** [comment]
> @omar-ocampo , could you look into comments from Makito, and rerun the PR without skipping any tests?
> Remove all CI commit pragma and just use the one below
> Test-tag-hw-small: pr,hw,small autotest

Done

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/container/autotest.py
Comment (shimizukko) [reply src/tests/ftest/container/autotest.py]: Sorry I said you can make them locally above. It wasn't clear.

**Comment by omar-ocampo** [comment]
All tests within hardware medium failed, some step after test run failed:
[2021-04-14T13:05:41.701Z] Recording test results
[2021-04-14T13:05:42.200Z] [Checks API] No suitable checks publisher found.
Functional_Hardware_Medium failed: 1[2021-04-14T13:05:43.566Z] File Delete Operation:
[2021-04-14T13:05:43.643Z] 0 files found for include pattern '*.memcheck.xml' and exclude pattern 'null'

**Comment by omar-ocampo** [comment]
@sylviachanoiyee  friendly reminder

**Comment by sylviachanoiyee** [comment]
@okulkarni , can you get another review for the PR?
The medium stage seems to failed archiving. 
Can you report the issue or open a new ticket? 
Can you retrigger the run?

**Comment by omar-ocampo** [comment]
> @okulkarni , can you get another review for the PR?
> The medium stage seems to failed archiving.
> Can you report the issue or open a new ticket?
> Can you retrigger the run?

Here is the ticket: https://jira.hpdd.intel.com/browse/DAOS-7245 

Retriggering seems more like coin toss hoping to not see the issue and should not be required, "All 6474 tests for this pipeline are passing".

**Comment by sylviachanoiyee** [comment]
> > @okulkarni , can you get another review for the PR?
> > The medium stage seems to failed archiving.
> > Can you report the issue or open a new ticket?
> > Can you retrigger the run?
> 
> Here is the ticket: https://jira.hpdd.intel.com/browse/DAOS-7245
> 
> Retriggering seems more like coin toss hoping to not see the issue and should not be required, "All 6474 tests for this pipeline are passing".

I think you need to fix the description in the ticket DAOS-7245. The failure is actually in files failed to archive. You will be able to to see the error in the fn hw med console. In fact, there might be existing ticket in JIRA related to files failing to archive if you search JIRA. If the issue is the same, you can re-open the ticket.

I'm not sure which test failed archiving but I cannot land a PR if it directly or indirectly trigger failure in file archiving and cause the stage to fail. That's why I'm requesting for a retrigger. It would also help if you can pin point the test that failed to archive and mention it in the ticket.

**Comment by omar-ocampo** [comment]
> > > @okulkarni , can you get another review for the PR?
> > > The medium stage seems to failed archiving.
> > > Can you report the issue or open a new ticket?
> > > Can you retrigger the run?
> > 
> > 
> > Here is the ticket: https://jira.hpdd.intel.com/browse/DAOS-7245
> > Retriggering seems more like coin toss hoping to not see the issue and should not be required, "All 6474 tests for this pipeline are passing".
> 
> I think you need to fix the description in the ticket DAOS-7245. The failure is actually in files failed to archive. You will be able to to see the error in the fn hw med console. In fact, there might be existing ticket in JIRA related to files failing to archive if you search JIRA. If the issue is the same, you can re-open the ticket.
> 
> I'm not sure which test failed archiving but I cannot land a PR if it directly or indirectly trigger failure in file archiving and cause the stage to fail. That's why I'm requesting for a retrigger. It would also help if you can pin point the test that failed to archive and mention it in the ticket.

Ticket updated.
The command failing is:
[2021-04-14T13:05:29.617Z] + SUITE=DAOS_Rebuild
[2021-04-14T13:05:29.617Z] + sed -i 's/case name/case classname="FTEST_daos_test.DAOS_Rebuild" name/' /var/tmp/ftest/avocado/job-results/daos_test/daos_core_test-rebuild.py/test-results/15-._daos_test_daos_core_test-rebuild.py_DaosCoreTestRebuild.test_rebuild_28_transport_config-args-daos_test-stopped_ranks-test_name-transport_config-hosts-pool-0-1-transport_config-timeouts-a3f9/data/DAOS_Rebuild_cmocka_results.xml
[2021-04-14T13:05:29.617Z] + exit 1

I'm not familiar with the archive step, but I do wonder what could be the relation between adding a test and the failure above, any ideas?  The test has also been retrigger.

**Comment by omar-ocampo** [comment]
@sylviachanoiyee , the previous error was not reported, however this test case failed:

Test Hardware / Functional_Hardware_Small / 1-./control/daos_admin_privileged.py:DaosAdminPrivTest.test_daos_admin_format;transport_config-transport_config-hosts-servers-transport_config-setup-1c72 – FTEST_control.DaosAdminPrivTest

It shouldnt affect the merge of this change.

**Comment by omar-ocampo** [comment]
@sylviachanoiyee, friendly reminder, how can I help to complete the review process?

**Review by rpadma2 — DISMISSED** [review]

**Comment by sylviachanoiyee** [comment]
@okulkarni , could you merge with master and kick off another build?

**Comment by mchaarawi** [comment]
> @okulkarni , could you merge with master and kick off another build?

I think you mean to tag @omar-ocampo 
omkar is not with us anymore ;-)

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:26]: This was request to be removed from init in another comment.

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:31]: No need.

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/container/autotest.py
Comment (phender) [inline src/tests/ftest/container/autotest.py:21]: This is not required and can be removed to simplify things.  The `self.daos_cmd` only needs to be a class attribute if it is accessed across multiple test methods.  As there is only one test method using it - `test_container_autotest()` it can be a local method variable.

File: src/tests/ftest/container/autotest.py
Comment (phender) [inline src/tests/ftest/container/autotest.py:31]:
```suggestion
        daos_cmd = self.get_daos_command()
```

File: src/tests/ftest/container/autotest.py
Comment (phender) [inline src/tests/ftest/container/autotest.py:34]: Besides a non-zero return code is there anything else we should be verifying from the `daos pool autotest` command?

File: src/tests/ftest/container/autotest.py
Comment (phender) [inline src/tests/ftest/container/autotest.py:26]: We should probably use the new tag specification format here:
```suggestion
        :avocado: tags=all,full_regression,daily_regression
        :avocado: tags=hw,small
        :avocado: tags=container,autotest,containerautotest
```

File: src/tests/ftest/container/autotest.py
Comment (phender) [inline src/tests/ftest/container/autotest.py:34]:
```suggestion
            daos_cmd.pool_autotest(pool=self.pool.uuid)
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:31]: Ok,  will edit

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:34]: The return code is enough.

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:26]: Will change

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:34]: Ok

**Review by phender — APPROVED** [review]

File: src/tests/ftest/container/autotest.py
Comment (shimizukko) [inline src/tests/ftest/container/autotest.py:21]: Is there any reason we have both full_regression and daily_regression? I think if you have daily, you wouldn't need full.

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/container/autotest.py
Comment (omar-ocampo) [reply src/tests/ftest/container/autotest.py:21]: I'm not 100% sure, I'll investigate it and I don't think this really blocks the merge.

**Comment by omar-ocampo** [comment]
@jolivier23 , can we merge this patch?

**Pull request merged**
